x/mongo/driver/operation.go

```diff
@@ -872,13 +892,25 @@ func (Operation) decompressWireMessage(wm []byte) ([]byte, error) {
 
 func (op Operation) createWireMessage(ctx context.Context, dst []byte,
 	desc description.SelectedServer, conn Connection) ([]byte, startedInformation, error) {
+	// Calculate value of 'maxTimeMS' field to append to the wire message based on the current
+	// context's deadline and the 90th percentile RTT if op.Timeout is set.
+	var maxTimeMS *int64
+	if op.Timeout != nil {
+		if deadline, ok := ctx.Deadline(); ok {
+			remainingTimeout := time.Until(deadline)
+			maxTimeMSVal := int64(remainingTimeout/time.Millisecond) -
+				int64(desc.RTT90/time.Millisecond)
+			maxTimeMS = &maxTimeMSVal
```

Comment:
How stale could the "RTT90" value pulled from the server description be here? Are there any issues that could happen if the "RTT90" value is very stale (e.g. 20+ seconds old)?

Reply:
Oh boy excellent catch. This _also_ needs to pull from `Server` and not from the server description that is only updated on handshakes. Updated. 

The value of `RTT90` could still be stale here. It _should_ be updated with every heartbeat, but with a default `heartbeatInterval` of 10s, 20+ seconds old is not inconceivable. The "amount" of staleness is no different than the staleness of `MinRTT` in the current driver design, though. I'll add testing in GODRIVER-2405 to assert that it does not get _too_ stale (maybe > 30s, will discuss offline).